src/inmanta/module.py

```diff
@@ -910,11 +918,13 @@ def set(cls, project: "Project") -> None:
         os.chdir(project._path)
         plugins.PluginMeta.clear()
         loader.unload_inmanta_plugins()
+        loader.PluginModuleFinder.configure_module_finder(cls._project.modulepath, modules_to_ignore=[])
 
     def load(self) -> None:
         if not self.loaded:
+            if not self.is_using_virtual_env():
+                self.use_virtual_env()
```

Comment:
Do I understand correctly that we hereby activate the compiler venv and never deactivate it? As a result any following `pip install` calls will install in the compiler venv?
(not an issue with your implementation per se, my pull request has the same behavior)

Reply:
Nevermind, `env.ProcessEnv` uses `sys.executable`, so this will always install in the encapsulating venv, not in the compiler venv.